pkg/tinyvcdiff/contrib/tinyvcdiff_mtd/mtd.c

```diff
@@ -0,0 +1,55 @@
+#include "vcdiff_mtd.h"
+
+static int erase (void *dev, size_t offset, size_t len)
+{
+    int rc = 0;
+    vcdiff_mtd_t *mtd = (vcdiff_mtd_t *) dev;
+    size_t addr = offset + len;
+    size_t page = addr / mtd->dev->page_size;
+    size_t sector = page / mtd->dev->pages_per_sector;
+    size_t cnt = sector - mtd->next_erase_sector + 1;
+
+    /* early exit if all sectors are already in the erased state */
+    if (cnt == 0) {
+        goto ret;
+    }
+
+    /* erase sectors */
+    rc = mtd_erase_sector(mtd->dev, mtd->next_erase_sector, cnt);
+    if (rc != 0) {
+        goto ret;
+    }
+
+    mtd->next_erase_sector += cnt;
+
+ret:
+    return rc;
+}
+
+static int read (void *dev, uint8_t *dest, size_t offset, size_t len)
```

Comment:
```suggestion
static int _read(void *dev, uint8_t *dest, size_t offset, size_t len)
```

You want to avoid clashing with [`read(2)`](https://man7.org/linux/man-pages/man2/read.2.html)